src/layers/transform/gather.cu

```diff
@@ -33,87 +33,107 @@ namespace lbann {
 namespace {
 
 using Dim2 = gpu_lib::array<size_t, 2>;
+using Dim3 = gpu_lib::array<size_t, 3>;
 
 /** @brief Kernel for scattering a 2D tensor along dim 1
  *
  *  output(j,indices(j,i)) = values(j,i)
  *
- *  Block dimensions: bdimx x bdimy x 1
+ *  Block dimensions: bdimx x bdimy x bdimz
  *
- *  Grid dimensions: (values_dim[1] / bdimx) x (values_dim[0] / bdimy) x 1
+ *  Grid dimensions: (num_columns_input_mat / bdimx) x (num_rows / bdimy) x mb_size /bdimz
```

Comment:
These grid dimensions aren't used when launching the kernel. It doesn't affect correctness, but will likely hurt performance.